assets/js/src/core/modules/element/tree/node/with-droppable/hooks/use-permission-checker.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import { checkElementPermission } from '@Pimcore/modules/element/permissions/permission-helper'
+import { type Element } from '@Pimcore/modules/element/element-helper'
+
+interface PermissionCheckerReturn {
+  isAllowedToDrop: (element: Element) => boolean
+}
+
+export const usePermissionChecker = (): PermissionCheckerReturn => {
```

Comment:
No need to change it - but in my opinion this should not be a use hook but a simple method (as there is no interaction with state/react)

Reply:
yeah, good point - we shouldn't use hooks if not needed - I converted the hook to a helper - if you have any better suggestions for the name I'm open to change that as well :)

Reply:
`isAllowedToMove` might be a slightly more precise name, but not that important ;)

Reply:
done with https://github.com/pimcore/studio-ui-bundle/pull/1019/commits/de2685f5160b3debc348210d63cc8aed083f43f9